website/files/utils.py

```diff
@@ -13,6 +13,7 @@ def copy_files(src, target_node, parent=None, name=None):
     cloned.parent = parent
     cloned.node = target_node
     cloned.name = name or cloned.name
+    cloned.archived_from_id = src._id
```

Comment:
Yes! This function is called when OSFS files are archived/copied -- there are similar functions in Waterbutler that are called for add-on files.